scripts/SD_converter.py

```diff
@@ -0,0 +1,127 @@
+    ----------
+    input_file : str
+        Path to the input file
+    output_file : str
+        Path to the output file
+    format : str
+        Output format, either "Jsonl" or "choices"
+
+    Methods
+    -------
+    convert():
+        Converts the input file to the specified output format and creates the output file.
+    """
+
+    def __init__(self, input_file: str, output_file: str, format: str):
+        self.input_file = input_file
+        self.output_file = output_file
+        self.format = format
+
+    def _parse_input_file(self) -> List[Tuple[str, str]]:
+        """
+        Parses the input file to extract questions with their respective answers.
+
+        Returns
+        -------
+        List[Tuple[str, str]]
+            The parsed question and answer pairs from the input file.
+        """
+        with open(self.input_file, "r") as file:
+            text = file.read()
+
+        entries = text.split("Question:")
+
+        qa_pairs = []
+        for entry in entries[1:]:
+            question, answer = entry.split("Answer:")
+            question = question.strip()
+            answer = answer.strip()
+            qa_pairs.append((answer, question))
+
+        return qa_pairs
+
+    def _to_jsonl(self, qa_pairs: List[Tuple[str, str]]) -> None:
+        """
+        Converts and writes the question and answer pairs to the output file in JSONL format.
+
+        Parameters
+        ----------
+        qa_pairs : List[Tuple[str, str]]
+            The question and answer pairs from the input file
+        """
+        jsonl_output = []
+        for answer, question in qa_pairs:
+            qa_pair = {"question": question, "answer": answer}
+            jsonl_output.append(json.dumps(qa_pair))
+
+        with open(self.output_file, "w") as f:
+            for line in jsonl_output:
+                f.write(f"{line}\n")
+
+    def _to_choices(self, qa_pairs: List[Tuple[str, str]]) -> None:
+        """
+        Converts and writes the question and answer pairs to the output file in the "choices" format.
+
+        Parameters
+        ----------
+        qa_pairs : List[Tuple[str, str]]
+            The question and answer pairs from the input file
+        """
+        choices = []
+        for answer, question in qa_pairs:
+            qa_pair = {"prompt": question, "choices": answer}
+            choices.append(json.dumps(qa_pair))
+
+        with open(self.output_file, "w") as f:
+            for line in choices:
+                f.write(f"{line}\n")
+
+    def convert(self) -> None:
+        """
+        Converts the input file to the specified output format and creates the output file.
+
+        Raises
+        ------
+        Exception
+            If input file doesn't exist, output file already exists, or an invalid format is provided.
+        """
+        if not os.path.isfile(self.input_file):
+            raise Exception("Input file does not exist!")
+
+        if os.path.isfile(self.output_file):
```

Comment:
Allow overwriting of existing files. If I change an input file, I want to regenerate the corresponding output file. Force users to delete it manually first is not convenient.